internal/helm-workload/helm_workload_instance.go

```diff
@@ -0,0 +1,239 @@
+	install.Namespace = fmt.Sprintf("%s-%s", *helmWorkloadInstance.Name, util.RandomAlphaString(10))
+	install.CreateNamespace = true
+	install.PostRenderer = &ThreeportPostRenderer{
+		HelmWorkloadDefinition: helmWorkloadDefinition,
+		HelmWorkloadInstance:   helmWorkloadInstance,
+	}
+	install.RepoURL = *helmWorkloadDefinition.HelmRepo
+	install.DependencyUpdate = true
+	chartPath, err := install.LocateChart(*helmWorkloadDefinition.HelmChart, settings)
+	if err != nil {
+		return 0, fmt.Errorf("failed to set helm chart path: %w", err)
+	}
+
+	// load the chart
+	chart, err := loader.Load(chartPath)
+	if err != nil {
+		return 0, fmt.Errorf("failed to load helm chart: %w", err)
+	}
+
+	// write the value files to merge as needed
+	var valueFiles []string
+	if helmWorkloadDefinition.HelmValuesDocument != nil {
+		filePath := fmt.Sprintf("/tmp/%d-definition-vals.yaml", (*helmWorkloadInstance.ID))
+		if err := os.WriteFile(
+			filePath,
+			[]byte(*helmWorkloadDefinition.HelmValuesDocument),
+			0644,
+		); err != nil {
+			return 0, fmt.Errorf("failed to values file for helm workload definition values: %w", err)
+		}
+		valueFiles = append(valueFiles, filePath)
+	}
+	if helmWorkloadInstance.HelmValuesDocument != nil {
+		filePath := fmt.Sprintf("/tmp/%d-instance-vals.yaml", (*helmWorkloadInstance.ID))
+		if err := os.WriteFile(
+			filePath,
+			[]byte(*helmWorkloadInstance.HelmValuesDocument),
+			0644,
+		); err != nil {
+			return 0, fmt.Errorf("failed to values file for helm workload instance values: %w", err)
+		}
+		valueFiles = append(valueFiles, filePath)
+	}
+
+	// merge the helm values
+	var helmValues map[string]interface{}
+	if len(valueFiles) > 0 {
+		vals := values.Options{ValueFiles: valueFiles}
+		helmVals, err := vals.MergeValues(nil)
+		if err != nil {
+			return 0, fmt.Errorf("failed to merge helm values: %s", err)
+		}
+		helmValues = helmVals
+	}
+
+	// deploy the helm workload
+	_, err = install.Run(chart, helmValues)
+	if err != nil {
+		return 0, fmt.Errorf("failed to install helm chart: %w", err)
+	}
+
+	return 0, nil
+}
+
+// helmWorkloadInstanceCreated reconciles state for a helm workload
+// instance when it is changed.
+func helmWorkloadInstanceUpdated(
+	r *controller.Reconciler,
+	helmWorkloadInstance *v0.HelmWorkloadInstance,
+	log *logr.Logger,
+) (int64, error) {
+	return 0, nil
+}
+
+// helmWorkloadInstanceCreated reconciles state for a helm workload
+// instance when it is removed.
+func helmWorkloadInstanceDeleted(
+	r *controller.Reconciler,
+	helmWorkloadInstance *v0.HelmWorkloadInstance,
+	log *logr.Logger,
+) (int64, error) {
+	// get helm action config
+	actionConf, _, err := getHelmActionConfig(r, helmWorkloadInstance)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get a helm action config: %w", err)
+	}
+
+	// Setting up uninstall action
+	uninstall := action.NewUninstall(actionConf)
+
+	// Running uninstall action
+	releaseName := fmt.Sprintf("%s-release", *helmWorkloadInstance.Name)
```

Comment:
put this into a function so that Create and Delete operations are always identicaly in creating the release name